### PR TITLE
feat(ec2): prefix list references in aws_route and aws_security_group_rule

### LINKED ISSUE #605: Add prefixListId ref and selector for aws_route and aws_security_group_rule
### What problem are you facing?

We need to resolve `aws_ec2_managed_prefix_list` reference in `aws_route` and `aws_security_group_rule` resources using ref and selector.

### PR DESCRIPTION
**Pull request by duizabojul**
### Description of your changes

Add new fields to resolve aws_ec2_managed_prefix_list in aws_route and aws_security_group_rule resources using ref and selector 

Fixes #605

I have:

- [ ] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Tested on our internal aws account

### CODE REVIEW

File: config/ec2/config.go [hunk 1]
```diff
@@ -237,6 +242,9 @@ func Configure(p *config.Provider) {
 		r.References["nat_gateway_id"] = config.Reference{
 			Type: "NATGateway",
 		}
+		r.References["destination_prefix_list_id"] = config.Reference{
+			Type: "ManagedPrefixList",
```
Comment (ulucinar) [inline config/ec2/config.go:246]: Please see the comment above:
```suggestion
			TerraformName: "aws_ec2_managed_prefix_list",
```

**Review by ulucinar — COMMENTED** [review]
Thank you @duizabojul for improving the API! Left some comments for you to consider.

**Comment by sergenyalcin** [comment]
`CI / ci / check-diff` job is failing. @duizabojul could you please rebase this branch?

**Comment by duizabojul** [comment]
Do you need an action from me?

**Comment by jeanduplessis** [comment]
/test-examples="examples/ec2/route.yaml"

**Comment by jeanduplessis** [comment]
/test-examples="examples/ec2/securitygrouprule.yaml"

**Comment by sergenyalcin** [comment]
It seems that, the `aws_security_group_rule` example has missing dependencies. Because of this missing dependencies, the uptest job is failing. `aws_security_group_rule` resource needs a SecurityGroup resource. So we may trigger the tests by also adding the `examples/ec2/securitygroup.yaml`. I hope this resolves the failure of uptest.

**Comment by sergenyalcin** [comment]
/test-examples="examples/ec2/securitygroup.yaml,examples/ec2/securitygrouprule.yaml"

**Comment by sergenyalcin** [comment]
https://github.com/upbound/provider-aws/actions/runs/4680973210 Uptest run is successful.

**Review by sergenyalcin — APPROVED** [review]
Thanks @duizabojul LGTM!

**Pull request merged**